utils_test.go

```diff
@@ -0,0 +1,23 @@
+package main
+
+import (
+	"fmt"
+	"testing"
+
+	dto "github.com/prometheus/client_model/go"
+)
+
+// Check if expected results are in the registry
+func checkRegistryResults(expRes map[string]float64, mfs []*dto.MetricFamily, t *testing.T) {
+	if len(mfs) == 0 {
+		t.Fatal("No metrics returned in registry")
+	}
+	for i := range mfs {
+		if _, ok := expRes[mfs[i].GetName()]; ok {
```

Comment:
You've still got the problem where if mfs was empty that this check would succeed